# Week 3

In [2]:
import pandas as pd

In [4]:
url = "https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M"

In [57]:
raw_data = pd.read_html(url)

In [63]:
df = raw_data[0]

In [67]:
df = df[df.Borough != "Not assigned"]

In [69]:
df.head()

Postal code           Borough                                  Neighborhood
2         M3A        North York                                     Parkwoods
3         M4A        North York                              Victoria Village
4         M5A  Downtown Toronto                    Regent Park / Harbourfront
5         M6A        North York             Lawrence Manor / Lawrence Heights
6         M7A  Downtown Toronto  Queen's Park / Ontario Provincial Government

In [62]:
df.shape

(103, 3)

In [74]:
df['Postal code'].value_counts()

M1T    1
M9C    1
M6G    1
M2H    1
M5V    1
      ..
M9N    1
M5W    1
M5E    1
M4V    1
M4H    1
Name: Postal code, Length: 103, dtype: int64

### The total number of rows in our data frame is 103. We can also see the all values in "Postal code" column are unique using the "value_counts()" because the length here is also 103

In [73]:
df['Neighborhood'].isnull().sum()

0

### Also there is no "Not assigned" row in the "Neighborhood" column

In [75]:
df.shape

(103, 3)